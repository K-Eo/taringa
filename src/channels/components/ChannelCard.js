import React from 'react'

import CardDecorator from '../../common/CardDecorator'

const ChannelCard = ({ channel, control }) => {
  return (
    <CardDecorator
      avatar={channel.thumbnail}
      cover={channel.background}
      placeholder={control.status === 'fetching'}
    >
      <div className="card-body">
        <h5 className="card-title">{channel.title}</h5>

        {channel.description !== channel.title && (
          <p className="card-text mt-3 mb-4">{channel.description}</p>
        )}

        <button className="btn btn-primary btn-block font-weight-bold mb-3">
          SEGUIR
        </button>

        <p className="text-muted small mb-0">
          Siguiendo este canal podrás ver sus publicaciones en tu página de
          inicio
        </p>
      </div>
    </CardDecorator>
  )
}

ChannelCard.defaultProps = {
  channel: {},
  control: { status: 'fetching' },
}

export default ChannelCard
