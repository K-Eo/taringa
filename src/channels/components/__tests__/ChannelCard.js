import React from 'react'

import ChannelCard from '../ChannelCard'

describe('Channel Card', () => {
  it('exists', () => {
    expect(ChannelCard).toBeDefined()
  })

  it('renders', () => {
    const control = {
      status: 'success',
    }

    const channel = {
      background: 'bar',
      description: 'lorem',
      thumbnail: 'foo',
      title: 'baz',
    }

    const wrapper = shallow(<ChannelCard control={control} channel={channel} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders placeholder', () => {
    const control = {
      status: 'fetching',
    }

    const wrapper = shallow(<ChannelCard control={control} />)

    expect(wrapper).toMatchSnapshot()
  })
})
