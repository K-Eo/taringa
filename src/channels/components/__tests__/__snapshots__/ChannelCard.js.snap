// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel Card renders 1`] = `
<CardDecorator
  avatar="foo"
  cover="bar"
  link={true}
  placeholder={false}
  to="/c/foo"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title text-truncate"
      title="baz"
    >
      baz
    </h5>
    <button
      className="btn btn-primary btn-block btn-sm font-weight-bold mt-4 mb-2"
    >
      SEGUIR
    </button>
    <p
      className="text-muted small mb-0"
    >
      Siguiendo este canal podr치s ver sus publicaciones en tu p치gina de inicio
    </p>
  </div>
</CardDecorator>
`;

exports[`Channel Card renders placeholder 1`] = `
<CardDecorator
  link={true}
  placeholder={true}
  to="/c/undefined"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title text-truncate"
    />
    <button
      className="btn btn-primary btn-block btn-sm font-weight-bold mt-4 mb-2"
    >
      SEGUIR
    </button>
    <p
      className="text-muted small mb-0"
    >
      Siguiendo este canal podr치s ver sus publicaciones en tu p치gina de inicio
    </p>
  </div>
</CardDecorator>
`;
